Program.cs

```diff
@@ -11,11 +16,74 @@
 // Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
 builder.Services.AddEndpointsApiExplorer();
 
+builder.Services.AddIdentity<IdentityUser, IdentityRole>()
+    .AddEntityFrameworkStores<RecipleaseContext>()
+    .AddDefaultTokenProviders();
+
+
+builder.Services.AddAuthentication(options =>
+    {
+        options.DefaultAuthenticateScheme = JwtBearerDefaults.AuthenticationScheme;
+        options.DefaultChallengeScheme = JwtBearerDefaults.AuthenticationScheme;
+        options.DefaultScheme = JwtBearerDefaults.AuthenticationScheme;
+    })
+    .AddJwtBearer(options =>
+    {
+        options.SaveToken = true;
+        options.RequireHttpsMetadata = false;
+        options.TokenValidationParameters = new TokenValidationParameters()
+        {
+            ValidateIssuer = true,
+            ValidateAudience = true,
+            ValidAudience = Environment.GetEnvironmentVariable("ValidAudience"),
+            ValidIssuer = Environment.GetEnvironmentVariable("ValidIssuer"),
+            IssuerSigningKey =
+                new SymmetricSecurityKey(Encoding.UTF8.GetBytes(Environment.GetEnvironmentVariable("Secret")))
+        };
+    })
+        .AddGoogle(googleOptions => 
+{
+    var clientId = Environment.GetEnvironmentVariable("GOOGLE_OAUTH_CLIENT_ID");
+    var clientSecret = Environment.GetEnvironmentVariable("GOOGLE_OAUTH_CLIENT_SECRET");
+
+    if (clientId == null)
+        throw new ArgumentNullException(nameof(clientId), "Google OAuth Client Id cannot be null.");
+    if (clientSecret == null) 
+        throw new ArgumentNullException(nameof(clientSecret), "Google OAuth Client Secret cannot be null.");
+    
+    googleOptions.ClientId = clientId;
+    googleOptions.ClientSecret = clientSecret;
+});
+
```

Comment:
indentation looks off.
```suggestion
    {
        var clientId = Environment.GetEnvironmentVariable("GOOGLE_OAUTH_CLIENT_ID");
        var clientSecret = Environment.GetEnvironmentVariable("GOOGLE_OAUTH_CLIENT_SECRET");

        if (clientId == null)
            throw new ArgumentNullException(nameof(clientId), "Google OAuth Client Id cannot be null.");
        if (clientSecret == null) 
            throw new ArgumentNullException(nameof(clientSecret), "Google OAuth Client Secret cannot be null.");
    
        googleOptions.ClientId = clientId;
        googleOptions.ClientSecret = clientSecret;
    });

```